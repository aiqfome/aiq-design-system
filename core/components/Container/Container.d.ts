import React, { ReactNode } from 'react';
interface TabProps {
    content: any;
    index: number;
    value?: number;
    active?: boolean;
    variant?: 'default' | 'contained' | 'card';
}
export interface Props {
    tabs?: Array<TabProps>;
    title?: string;
    tabIndex?: number;
    header?: ReactNode;
    children?: ReactNode;
    containerProps?: any;
    tabsExtra?: ReactNode;
    onChangeTab?: (event: any, newValue: any) => void;
}
export declare const Container: React.FC<Props>;
export {};
//# sourceMappingURL=Container.d.ts.map