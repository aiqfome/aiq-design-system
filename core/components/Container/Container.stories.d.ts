import React from 'react';
declare const _default: {
    component: React.FC<import("./Container").Props>;
    title: string;
};
export default _default;
export declare const Basic: React.FC;
export declare const WithTitle: React.FC;
export declare const WithHeader: React.FC;
export declare const WithTabs: React.FC;
//# sourceMappingURL=Container.stories.d.ts.map