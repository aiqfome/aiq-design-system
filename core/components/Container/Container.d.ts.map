{"version":3,"file":"Container.d.ts","sourceRoot":"","sources":["../../../src/components/Container/Container.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,SAAS,EAAE,MAAM,OAAO,CAAA;AAQxC,UAAU,QAAQ;IAChB,OAAO,EAAE,GAAG,CAAA;IACZ,KAAK,EAAE,MAAM,CAAA;IACb,KAAK,CAAC,EAAE,MAAM,CAAA;IACd,MAAM,CAAC,EAAE,OAAO,CAAA;IAChB,OAAO,CAAC,EAAE,SAAS,GAAG,WAAW,GAAG,MAAM,CAAA;CAC3C;AAED,MAAM,WAAW,KAAK;IACpB,IAAI,CAAC,EAAE,KAAK,CAAC,QAAQ,CAAC,CAAA;IACtB,KAAK,CAAC,EAAE,MAAM,CAAA;IACd,QAAQ,CAAC,EAAE,MAAM,CAAA;IACjB,MAAM,CAAC,EAAE,SAAS,CAAA;IAClB,QAAQ,CAAC,EAAE,SAAS,CAAA;IACpB,cAAc,CAAC,EAAE,GAAG,CAAA;IACpB,SAAS,CAAC,EAAE,SAAS,CAAA;IACrB,WAAW,CAAC,EAAE,CAAC,KAAK,EAAE,GAAG,EAAE,QAAQ,EAAE,GAAG,KAAK,IAAI,CAAA;CAClD;AAwED,eAAO,MAAM,SAAS,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAwCrC,CAAA"}