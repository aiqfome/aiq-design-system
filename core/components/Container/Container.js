"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Container = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const Divider_1 = require("../Divider");
const Flex_1 = require("../Flex");
const Text_1 = require("../Text");
const Tab_1 = require("../Tab");
const StyledContainer = (_a) => {
    var { children, tabIndex, tabs = [] } = _a, props = __rest(_a, ["children", "tabIndex", "tabs"]);
    let border = '12px';
    if (tabs.length && tabIndex === 0) {
        border = '0 12px 12px 12px';
    }
    return (react_1.default.createElement(Flex_1.Flex, Object.assign({ borderRadius: border, flexDirection: 'column', backgroundColor: 'white', border: '1px solid lightGrey', mt: tabs.length && '-1px' }, props), children));
};
const ContainerWrapper = (_a) => {
    var { title, header, children } = _a, props = __rest(_a, ["title", "header", "children"]);
    if (title) {
        return (react_1.default.createElement(StyledContainer, Object.assign({ "data-testid": 'container' }, props),
            react_1.default.createElement(Text_1.Text, { "data-testid": 'container-text', color: 'almostBlack', fontWeight: 'medium', fontSize: 'xxlarge', p: 10 }, title),
            react_1.default.createElement(Divider_1.Divider, null),
            children));
    }
    if (header) {
        return (react_1.default.createElement(StyledContainer, Object.assign({ "data-testid": 'container' }, props),
            header,
            react_1.default.createElement(Divider_1.Divider, null),
            children));
    }
    return (react_1.default.createElement(StyledContainer, Object.assign({ "data-testid": 'container' }, props), children));
};
exports.Container = (_a) => {
    var { children, tabIndex, tabs = [], tabsExtra, onChangeTab, containerProps } = _a, props = __rest(_a, ["children", "tabIndex", "tabs", "tabsExtra", "onChangeTab", "containerProps"]);
    return (react_1.default.createElement(Flex_1.Flex, Object.assign({ flexDirection: 'column' }, containerProps),
        tabs.length > 0 && (react_1.default.createElement(Tab_1.Tabs, { mr: '10px', pr: '15px', "data-testid": 'container-tab', variant: 'card', extra: tabsExtra, onChange: onChangeTab }, tabs.map((tab, index) => (react_1.default.createElement(Tab_1.Tab, Object.assign({}, tab, { "data-testid": 'container-tab-item', variant: 'card', value: tabIndex, key: tab.index || index, index: tab.index || index }), tab.content))))),
        react_1.default.createElement(ContainerWrapper, Object.assign({ tabIndex: tabIndex, tabs: tabs }, props), children)));
};
StyledContainer.propTypes = {
    tabs: prop_types_1.default.array,
    children: prop_types_1.default.node,
    tabIndex: prop_types_1.default.number
};
exports.Container.propTypes = {
    tabs: prop_types_1.default.array,
    header: prop_types_1.default.node,
    title: prop_types_1.default.string,
    tabsExtra: prop_types_1.default.any,
    children: prop_types_1.default.node,
    tabIndex: prop_types_1.default.number,
    onChangeTab: prop_types_1.default.func,
    containerProps: prop_types_1.default.object
};
ContainerWrapper.propTypes = {
    tabs: prop_types_1.default.array,
    header: prop_types_1.default.node,
    title: prop_types_1.default.string,
    children: prop_types_1.default.node
};
