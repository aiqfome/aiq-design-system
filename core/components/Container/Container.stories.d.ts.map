{"version":3,"file":"Container.stories.d.ts","sourceRoot":"","sources":["../../../src/components/Container/Container.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAiC,MAAM,OAAO,CAAA;;;;;AAUrD,wBAGC;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAMzB,CAAA;AAED,eAAO,MAAM,SAAS,EAAE,KAAK,CAAC,EAM7B,CAAA;AAED,eAAO,MAAM,UAAU,EAAE,KAAK,CAAC,EAuB9B,CAAA;AAED,eAAO,MAAM,QAAQ,EAAE,KAAK,CAAC,EAsC5B,CAAA"}