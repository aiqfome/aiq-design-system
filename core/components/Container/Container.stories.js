"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WithTabs = exports.WithHeader = exports.WithTitle = exports.Basic = void 0;
const react_1 = __importStar(require("react"));
const Button_1 = require("../Button");
const Flex_1 = require("../Flex");
const Input_1 = require("../Input");
const Text_1 = require("../Text");
const Container_1 = require("./Container");
const Tab_1 = require("../Tab");
exports.default = {
    component: Container_1.Container,
    title: 'Container'
};
exports.Basic = () => (react_1.default.createElement(Container_1.Container, null,
    react_1.default.createElement(Flex_1.Flex, { variant: 'centralized', p: 15 },
        react_1.default.createElement(Text_1.Text, null, "my content (:"))));
exports.WithTitle = () => (react_1.default.createElement(Container_1.Container, { title: 'Just a container!' },
    react_1.default.createElement(Flex_1.Flex, { variant: 'centralized', p: 15 },
        react_1.default.createElement(Text_1.Text, null, "my content (:"))));
exports.WithHeader = () => {
    const Header = () => (react_1.default.createElement(Flex_1.Flex, { justifyContent: 'space-between', p: 10 },
        react_1.default.createElement(Text_1.Text, { color: 'almostBlack', fontSize: 'xxlarge' }, "Just a title"),
        react_1.default.createElement(Flex_1.Flex, null,
            react_1.default.createElement(Input_1.Input, { placeholder: 'value' }),
            react_1.default.createElement(Button_1.Button, { ml: 5, variant: 'contained', palette: 'primary' }, "Filtrar"))));
    return (react_1.default.createElement(Container_1.Container, { header: react_1.default.createElement(Header, null) },
        react_1.default.createElement(Flex_1.Flex, { variant: 'centralized', p: 15 },
            react_1.default.createElement(Text_1.Text, null, "my content (:"))));
};
exports.WithTabs = () => {
    const [tabIndex, setTabIndex] = react_1.useState(0);
    const tabs = [
        {
            index: 0,
            content: 'pizza'
        },
        {
            index: 1,
            content: 'burguer'
        }
    ];
    const onChange = (_, index) => {
        setTabIndex(index);
    };
    return (react_1.default.createElement(Container_1.Container, { tabs: tabs, tabIndex: tabIndex, onChangeTab: onChange, tabsExtra: 'conte\u00FAdo extra' },
        react_1.default.createElement(Tab_1.TabPanel, { value: tabIndex, index: 0 },
            react_1.default.createElement(Flex_1.Flex, { variant: 'centralized', flex: 1, p: 15 },
                react_1.default.createElement(Text_1.Text, null, "aqui tem pizza"))),
        react_1.default.createElement(Tab_1.TabPanel, { value: tabIndex, index: 1 },
            react_1.default.createElement(Flex_1.Flex, { variant: 'centralized', flex: 1, p: 15 },
                react_1.default.createElement(Text_1.Text, null, "aqui tem v\u00E1rios burguers")))));
};
