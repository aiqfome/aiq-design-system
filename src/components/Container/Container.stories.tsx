import React, { ReactElement, useState } from 'react'

import { Button } from '../Button'
import { Flex } from '../Flex'
import { Input } from '../Input'
import { Text } from '../Text'

import { Container } from './Container'
import { TabPanel } from '../Tab'

export default {
  component: Container,
  title: 'Container'
}

export const Basic: React.FC = (): ReactElement => (
  <Container>
    <Flex variant='centralized' p={15}>
      <Text>my content (:</Text>
    </Flex>
  </Container>
)

export const WithTitle: React.FC = (): ReactElement => (
  <Container title='Just a container!'>
    <Flex variant='centralized' p={15}>
      <Text>my content (:</Text>
    </Flex>
  </Container>
)

export const WithHeader: React.FC = (): ReactElement => {
  const Header: React.FC = (): ReactElement => (
    <Flex justifyContent='space-between' p={10}>
      <Text color='almostBlack' fontSize='xxlarge'>
        Just a title
      </Text>
      <Flex>
        <Input placeholder='value' />

        <Button ml={5} variant='contained' palette='primary'>
          Filtrar
        </Button>
      </Flex>
    </Flex>
  )

  return (
    <Container header={<Header />}>
      <Flex variant='centralized' p={15}>
        <Text>my content (:</Text>
      </Flex>
    </Container>
  )
}

export const WithTabs: React.FC = (): ReactElement => {
  const [tabIndex, setTabIndex] = useState(0)

  const tabs = [
    {
      index: 0,
      content: 'pizza'
    },
    {
      index: 1,
      content: 'burguer'
    }
  ]

  const onChange = (_, index) => {
    setTabIndex(index)
  }

  return (
    <Container
      tabs={tabs}
      tabIndex={tabIndex}
      onChangeTab={onChange}
      tabsExtra='conteúdo extra'
    >
      <TabPanel value={tabIndex} index={0}>
        <Flex variant='centralized' flex={1} p={15}>
          <Text>aqui tem pizza</Text>
        </Flex>
      </TabPanel>

      <TabPanel value={tabIndex} index={1}>
        <Flex variant='centralized' flex={1} p={15}>
          <Text>aqui tem vários burguers</Text>
        </Flex>
      </TabPanel>
    </Container>
  )
}
