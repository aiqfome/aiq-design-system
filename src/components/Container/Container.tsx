import React, { ReactNode } from 'react'
import PropTypes from 'prop-types'

import { Divider } from '../Divider'
import { Flex } from '../Flex'
import { Text } from '../Text'
import { Tabs, Tab } from '../Tab'

interface TabProps {
  content: any
  index: number
  value?: number
  active?: boolean
  variant?: 'default' | 'contained' | 'card'
}

export interface Props {
  tabs?: Array<TabProps>
  title?: string
  tabIndex?: number
  header?: ReactNode
  children?: ReactNode
  onChangeTab?: (event: any, newValue: any) => void
}

const StyledContainer: React.FC<Props> = ({
  children,
  tabIndex,
  tabs = [],
  ...props
}) => {
  let border = '12px'
  if (tabs.length && tabIndex === 0) {
    border = '0 12px 12px 12px'
  }

  return (
    <Flex
      borderRadius={border}
      flexDirection='column'
      backgroundColor='white'
      border='1px solid lightGrey'
      mt={tabs.length && '-1px'}
      {...props}
    >
      {children}
    </Flex>
  )
}

const ContainerWrapper: React.FC<Props> = ({
  title,
  header,
  children,
  ...props
}) => {
  if (title) {
    return (
      <StyledContainer {...props}>
        <Text color='almostBlack' fontWeight='medium' fontSize='xxlarge' p={10}>
          {title}
        </Text>

        <Divider />

        {children}
      </StyledContainer>
    )
  }

  if (header) {
    return (
      <StyledContainer {...props}>
        {header}

        <Divider />

        {children}
      </StyledContainer>
    )
  }

  return <StyledContainer {...props}>{children}</StyledContainer>
}

export const Container: React.FC<Props> = ({
  children,
  tabIndex,
  tabs = [],
  onChangeTab,
  ...props
}) => {
  return (
    <>
      {tabs.length > 0 && (
        <Tabs
          mr='10px'
          variant='card'
          value={tabIndex || 0}
          onChange={onChangeTab}
        >
          {tabs.map((tab, index) => (
            <Tab
              {...tab}
              variant='card'
              value={tabIndex}
              key={tab.index || index}
              index={tab.index || index}
            >
              {tab.content}
            </Tab>
          ))}
        </Tabs>
      )}

      <ContainerWrapper tabIndex={tabIndex} tabs={tabs} {...props}>
        {children}
      </ContainerWrapper>
    </>
  )
}

StyledContainer.propTypes = {
  tabs: PropTypes.array,
  children: PropTypes.node,
  tabIndex: PropTypes.number
}

Container.propTypes = {
  tabs: PropTypes.array,
  header: PropTypes.node,
  title: PropTypes.string,
  children: PropTypes.node,
  tabIndex: PropTypes.number,
  onChangeTab: PropTypes.func
}

ContainerWrapper.propTypes = {
  tabs: PropTypes.array,
  header: PropTypes.node,
  title: PropTypes.string,
  children: PropTypes.node
}
