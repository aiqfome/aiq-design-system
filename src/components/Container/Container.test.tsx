import React from 'react'

import { render } from '../utils/test/render'

import { Container } from './Container'
import { fireEvent } from '@testing-library/react'

describe('Container', () => {
  it('should have to render without crashing', () => {
    const component = render(<Container />)
    expect(component).toBeTruthy()
  })

  it('should have a content when have a children', () => {
    const content = 'My Content'
    const { getByTestId } = render(<Container>{content}</Container>)

    expect(getByTestId('container')).toHaveTextContent(content)
  })

  it('should have a title when have prop title', () => {
    const content = 'My Content'
    const { getByTestId } = render(
      <Container title='My title'>{content}</Container>
    )

    const textContainer = getByTestId('container-text')

    expect(textContainer).toBeTruthy()
    expect(textContainer).toHaveTextContent('My title')
  })

  it('should have a header when have prop header', () => {
    const Header = () => <span data-testid='container-header'>TESTE</span>
    const content = 'My Content'
    const { getByTestId } = render(
      <Container header={<Header />}>{content}</Container>
    )

    const header = getByTestId('container-header')
    expect(header).toBeTruthy()
  })

  it('should have the same amount tabs of the array', () => {
    const tabs = [
      {
        index: 0,
        content: 'pizza'
      },
      {
        index: 1,
        content: 'burguer'
      }
    ]

    const { getAllByTestId } = render(<Container tabs={tabs} tabIndex={0} />)
    const tabsItem = getAllByTestId('container-tab-item')
    expect(tabsItem.length).toBe(tabs.length)
  })

  it('should have call onChangeTab when click on tab and prop tabs does not null', () => {
    const handleOnchange = jest.fn()
    const tabs = [
      {
        index: 0,
        content: 'pizza'
      },
      {
        index: 1,
        content: 'burguer'
      }
    ]

    const { getAllByTestId } = render(
      <Container tabs={tabs} tabIndex={0} onChangeTab={handleOnchange} />
    )
    const tabsItem = getAllByTestId('container-tab-item')
    fireEvent.click(tabsItem[0])
    expect(handleOnchange).toHaveBeenCalled()
  })

  it('should have a tab extra when have prop tabsExtra and tabs', () => {
    const tabsExtra = 'Test tabsExtra'

    const tabs = [
      {
        index: 0,
        content: 'pizza'
      },
      {
        index: 1,
        content: 'burguer'
      }
    ]

    const { getByTestId } = render(
      <Container
        tabs={tabs}
        tabIndex={0}
        title='My title'
        tabsExtra={tabsExtra}
      />
    )

    const container = getByTestId('container-tab')

    expect(container).toHaveTextContent(tabsExtra)
  })
})
