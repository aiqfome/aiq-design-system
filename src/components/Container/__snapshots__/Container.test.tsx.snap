// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-bdnylx dmpbKf"
>
  <div
    className="sc-bdnylx czbSZD"
  />
</div>
`;
